src/admin-copy-user-data.ts

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable no-await-in-loop,no-console */
+/**
+ * TPM ONLY SCRIPT! BE CAREFUL COPYING TO PROD!
+ * Script to copy data from one user on production or dev to another user on dev.
+ * Requires service accounts for database.
+ * serviceAccount.json (if using dev) and serviceAccountProd.json (if using prod) must be at the root.
+ *
+ * From root, run: `npm run ts-node -- src/admin-copy-user-data.ts <FROM_ENV>/<FROM_USER> <TO_ENV>/<TO_USER> <EXECUTE>`
+ * FROM_ENV and TO_ENV should be either "dev" or "prod"
+ * EXECUTE must be TRUE or FALSE. EXECUTE=FALSE will preview the changes, EXECUTE=TRUE will do the changes
+ * EXAMPLE: `npm run ts-node -- src/admin-copy-user-data.ts prod/noschiff.dev@gmail.com dev/nps39@cornell.edu TRUE`
+ */
+
+import { cert, initializeApp } from 'firebase-admin/app';
+import { getFirestore } from 'firebase-admin/firestore';
+
+const collections = [
+  'user-name',
+  'user-semesters',
+  'user-toggleable-requirement-choices',
+  'user-overridden-fulfillment-choices',
+  'user-selectable-requirement-choices',
+  'user-subject-colors',
+  'user-unique-incrementer',
+  'user-onboarding-data',
+];
+
+const fromArg = process.argv[2];
+const toArg = process.argv[3];
+const executeArg = process.argv[4];
+
+if (fromArg && toArg && (executeArg === 'TRUE' || executeArg === 'FALSE')) {
+  const [FROM_ENV, FROM] = fromArg.split('/');
+  const [TO_ENV, TO] = toArg.split('/');
+  const EXECUTE = executeArg === 'TRUE';
+  execute(FROM, FROM_ENV, TO, TO_ENV, EXECUTE).then(copied => {
+    if (EXECUTE) console.log(`Copied: [${copied}] from ${fromArg} to ${toArg}`);
+    else console.log(copied);
+  });
+} else {
+  throw new Error('Refer to the documentation to correctly run this script.');
+}
+
+async function execute(
+  FROM: string,
+  FROM_ENV: string,
+  TO: string,
+  TO_ENV: string,
+  EXECUTE: boolean
```

Comment:
I think it would be better to pass in an options object as opposed to multiple arguments because we have multiple string parameters.

Reply:
I narrowed with unions. Is a whole object necessary?